### PR TITLE
[#1782] Fixes of probe system

### PR DESCRIPTION
**Pull request by danielkryska**
Based on @Marcelinna comments in #1814 

- [Fix] Single user action should be recorded once (prevent HTML bubbling) 
- Add attributes to not mentioned items in ['Recommendation's probe placement ...'](https://docs.cyfronet.pl/display/FID/Recommendation%27s+probe+placement+and+specification+in+MP) to keep consistency
- Track user typing in catalog search, add listener to it

### CODE REVIEW

**Comment by Marcelinna** [comment]
lack of user action recordering here
image

**Comment by danielkryska** [comment]
Implementation of `ask provider` and trash icon in comparison are difficult to make. Modal is generic and should be re-implemented and triggering the icon is somehow blocked by stimulus action.

**Review by Marcelinna — APPROVED** [review]

**Review by wujuu — APPROVED** [review]

File: app/views/services/_service.html.haml [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
     service_path(service, category.present? ? { fromc: category.slug } : nil)
   - content_for :checkbox do
-    .mt-3.compare{"data-probe" => ""}
+    .mt-3.compare
       %label{ options(service.slug, comparison_enabled) }
         = check_box_tag "comparison", service.slug, checked?(service.slug), id: "comparison-#{service.id}",
                 class: "form-check-input",
                 disabled: !checked?(service.slug) && comparison_enabled,
-                data: { "target": "comparison.checkbox", "action": "click->comparison#update" }
+                data: { "target": "comparison.checkbox", "action": "click->comparison#update", "probe" => "" }
```
[the remark this answers is not in the document]
Comment (danielkryska) [reply app/views/services/_service.html.haml:11]: In HAML such code
```
data: { "target": "comparison.checkbox", "action": "click->comparison#update", "probe" => "" }
```

means this
```
"data-target": "comparison.checkbox", "data-action": "click->comparison#update", "data-probe" => ""

**Comment by goreck888** [comment]
@danielkryska install overcommit on your local instance because you don't check rubocop and linters warnings. That's the way your PRs tests are failing.

**Review by michal-szostak — APPROVED** [review]

**Pull request merged**